source/1_engine/rex_engine/src/string/stringentry.cpp

```diff
@@ -0,0 +1,97 @@
+#include "rex_engine/string/stringentry.h"
+
+#include <rex_std/memory.h>
+
+namespace rex
+{
+  //-------------------------------------------------------------------------
+  StringEntry::StringEntry(rsl::string_view characters) noexcept
+      : m_characters()
+  {
+    m_characters.reset(static_cast<char*>(malloc(characters.size() + 1))); // NOLINT(cppcoreguidelines-no-malloc)
+
+    rsl::memset(m_characters.get(), 0, characters.size() + 1);
+    rsl::memcpy(m_characters.get(), characters.data(), characters.size());
+  }
+  //-------------------------------------------------------------------------
+  StringEntry::StringEntry(StringEntry&& other) noexcept
+      : m_characters(rsl::exchange(other.m_characters, {}))
+  {
+  }
+
+  //-------------------------------------------------------------------------
+  StringEntry::~StringEntry() = default;
+
+  //-------------------------------------------------------------------------
+  rex::StringEntry& StringEntry::operator=(StringEntry&& other) noexcept
+  {
+    m_characters = rsl::exchange(other.m_characters, {});
+
+    return *this;
+  }
+
+  //-------------------------------------------------------------------------
+  rsl::string_view StringEntry::characters() const
+  {
+    return rsl::string_view(m_characters.get());
```

Comment:
provide the length as well, this saves a strlen call in the ctor
general rule is, if you have the length, pass it in